org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -0,0 +1,187 @@
+     */
+    getName() {
+        return this._name;
+    }
+
+    /**
+     * Gets device details (all fields), response formatted per http://docs.winkapiv2.apiary.io/
+     */
+    getDeviceDetailsAsync(deviceType, deviceId) {
+
+        // build request URI
+        var requestPath = '/' + deviceType + '/' + deviceId;
+
+        // Make the async request
+        return _makeRequest(requestPath, 'GET');
+    }
+
+    /**
+     * Puts device details (all fields) payload formatted per http://docs.winkapiv2.apiary.io/
+    */
+    putDeviceDetailsAsync(deviceType, deviceId, putPayload) {
+
+        // build request path and body
+        var requestPath = '/' + deviceType + '/' + deviceId;
+        var putPayloadString = JSON.stringify(putPayload);
+
+        // Make the async request
+        return _makeRequest(requestPath, 'PUT', putPayloadString);
+    }
+
+    /** 
+     * Given the hub specific device, returns the opent2t schema and translator
+    */
+    _getOpent2tInfo(winkDevice) {
+        if (winkDevice.thermostat_id != undefined) {
+            return { 
+                "schema": 'org.opent2t.sample.thermostat.superpopular',
+                "translator": "com-wink-thermostat"
+            };
+        }
+        else if (winkDevice.binary_switch_id != undefined) {
+            return { 
+                "schema": 'org.opent2t.sample.binaryswitch.superpopular',
+                "translator": "com-wink-binaryswitch"
+            };
+        }
+        
+        return undefined;
+    }
+
+    /**
+     * Internal helper method which makes the actual request to the wink service
+     */
+    _makeRequest(path, method, content) {
+        // build request URI
+        var requestUri = this._baseUrl + path;
+
+        // Set the headers
+        var headers = {
+            'Authorization': 'Bearer ' + this._accessToken.accessToken,
+            'Accept': 'application/json'
+        }
+
+        if (content) {
+            headers['Content-Type'] = 'application/json';
+            headers['Content-length'] = content.length;
+        }
+
+        var options = {
+            url: requestUri,
+            method: method,
+            headers: headers,
+            followAllRedirects: true
+        };
+
+        if (content) {
+            options.body = content;
+        }
+
+        // Start the async request
+        return request(options)
+            .then(function (body) {
+                return JSON.parse(body);
+            })
+            .catch(function (err) {
+                console.log("Request failed to: " + options.method + " - " + options.url);
+                console.log("Error            : " + err.statusCode + " - " + err.response.statusMessage);
+                // todo auto refresh in specific cases, issue 74
+                throw err;
+            });
+    }
+
+    /**
+     * Wink has a semi-complicated way of getting the actual device id of a device.
+     */
+    _getDeviceId(winkDevice) {
```

Comment:
Just as a general note (not a problem for this PR) this is specific to the v2 API. See issue #40 and after that we should not need to do this assuming the caller is on the v2 API.